daheinze-oscheer/hello/hello_test.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import "testing"
+
+func TestHello(t *testing.T) {
+
+	assertCorrectMessage := func(t *testing.T, got, want string) {
+		t.Helper()
+		if got != want {
+			t.Errorf("got %q want %q", got, want)
+		}
+	}
+
+	t.Run("saying hello to people", func(t *testing.T) {
+		got := Hello("Daniel", "")
+		want := "Hello, Daniel"
+		assertCorrectMessage(t, got, want)
+	})
+
+	t.Run("Saying 'Hello, World' when an empty string is supplied", func(t *testing.T) {
+		got := Hello("", "")
+		want := "Hello, World"
+		assertCorrectMessage(t, got, want)
+	})
+
+	t.Run("in Spanish", func(t *testing.T) {
+		got := Hello("Elodie", "Spanish")
+		want := "Hola, Elodie"
+		assertCorrectMessage(t, got, want)
+	})
+
+	t.Run("in French", func(t *testing.T) {
+		got := Hello("Francis", "French")
+		want := "Bonjour, Francis"
+		assertCorrectMessage(t, got, want)
+	})
```

Comment:
We can apply a for loop here to save some space and make it cleaner. Let's talk about this in our session later